util/webhook/webhook.go

```diff
@@ -44,429 +46,203 @@ const usernameRegex = `[a-zA-Z0-9_\.][a-zA-Z0-9_\.-]{0,30}[a-zA-Z0-9_\.\$-]?`
 	repoRegexp, err := regexp.Compile(regexpStr)
 	if err != nil {
-		return nil, fmt.Errorf("failed to compile regexp for repoURL '%s'", webURL)
+		return nil, fmt.Errorf("failed to compile regexp for repoURL '%s'", originalUrl)
 	}
 
 	return repoRegexp, nil
 }
 
-func (a *ArgoCDWebhookHandler) storePreviouslyCachedManifests(app *v1alpha1.Application, change changeInfo, trackingMethod string, appInstanceLabelKey string) error {
-	err := argo.ValidateDestination(context.Background(), &app.Spec.Destination, a.db)
-	if err != nil {
-		return fmt.Errorf("error validating destination: %w", err)
-	}
-
-	var clusterInfo v1alpha1.ClusterInfo
-	err = a.serverCache.GetClusterInfo(app.Spec.Destination.Server, &clusterInfo)
-	if err != nil {
-		return fmt.Errorf("error getting cluster info: %w", err)
-	}
-
-	var sources v1alpha1.ApplicationSources
-	if app.Spec.HasMultipleSources() {
-		sources = app.Spec.GetSources()
-	} else {
-		sources = append(sources, app.Spec.GetSource())
-	}
-
-	refSources, err := argo.GetRefSources(context.Background(), sources, app.Spec.Project, a.db.GetRepository, []string{}, false)
-	if err != nil {
-		return fmt.Errorf("error getting ref sources: %w", err)
-	}
-	source := app.Spec.GetSource()
-	cache.LogDebugManifestCacheKeyFields("moving manifests cache", "webhook app revision changed", change.shaBefore, &source, refSources, &clusterInfo, app.Spec.Destination.Namespace, trackingMethod, appInstanceLabelKey, app.Name, nil)
-
-	if err := a.repoCache.SetNewRevisionManifests(change.shaAfter, change.shaBefore, &source, refSources, &clusterInfo, app.Spec.Destination.Namespace, trackingMethod, appInstanceLabelKey, app.Name, nil); err != nil {
-		return fmt.Errorf("error setting new revision manifests: %w", err)
-	}
-
-	return nil
-}
-
-func sourceRevisionHasChanged(source v1alpha1.ApplicationSource, revision string, touchedHead bool) bool {
-	targetRev := parseRevision(source.TargetRevision)
-	if targetRev == "HEAD" || targetRev == "" { // revision is head
-		return touchedHead
-	}
-	targetRevisionHasPrefixList := []string{"refs/heads/", "refs/tags/"}
-	for _, prefix := range targetRevisionHasPrefixList {
-		if strings.HasPrefix(source.TargetRevision, prefix) {
-			return revision == targetRev
-		}
-	}
-
-	return source.TargetRevision == revision
+func GetWebUrlRegex(originalUrl string) (*regexp.Regexp, error) {
+	return getUrlRegex(originalUrl, `(?i)^(https?://|%[1]s@|ssh://(%[1]s@)?((alt)?ssh\.)?)%[2]s(:[0-9]+)?[:/]%[3]s$`)
 }
 
-func sourceUsesURL(source v1alpha1.ApplicationSource, webURL string, repoRegexp *regexp.Regexp) bool {
-	if !repoRegexp.MatchString(source.RepoURL) {
-		log.Debugf("%s does not match %s", source.RepoURL, repoRegexp.String())
-		return false
-	}
-
-	log.Debugf("%s uses repoURL %s", source.RepoURL, webURL)
-	return true
+func GetApiUrlRegex(originalUrl string) (*regexp.Regexp, error) {
+	return getUrlRegex(originalUrl, `(?i)^https?://%[2]s(:[0-9]+)?/?$`)
```

Comment:
The original regex used for this ([here](https://github.com/argoproj/argo-cd/blob/6dc559a/applicationset/webhook/webhook.go#L257)) allowed for SSH URLs which, AFAIK, aren’t valid for an API URL. If I’m mistaken on that, I can easily expand this to account for that. (There were no tests for this regex before. I added basic positive and negative cases in new `Test_GetApiUrlRegex`.)